tests/test_GLMestimatesingletrial.m

```diff
@@ -0,0 +1,68 @@
+function test_suite = test_GLMestimatesingletrial %#ok<*STOUT>
+
+  try % assignment of 'localfunctions' is necessary in Matlab >= 2016
+    test_functions = localfunctions(); %#ok<*NASGU>
+  catch % no problem; early Matlab versions can use initTestSuite fine
+  end
+
+  initTestSuite;
+
+end
+
+function test_GLMestimatesingletrial_system()
+
+  % "end-to-end" test of GLMestimatesingletrial
+  % only checks the HRF index of 400 voxels
+
+  [data, expected, output_dir] = set_up_test();
+
+  % GIVEN
+  design = data.design(1:3);
+  stimdur = data.stimdur;
+  tr = data.tr;
+  data = cellfun(@(x) x(51:70, 8:27, 1, :), data.data(1:3), 'UniformOutput', 0);
+
+  % WHEN
+  results = GLMestimatesingletrial(design, ...
+                                   data, ...
+                                   stimdur, ...
+                                   tr, ...
+                                   output_dir, ...
+                                   struct('wantmemoryoutputs', [1 1 1 1]));
+
+  % THEN
+  assertEqual(results{2}.HRFindex, expected{2}.HRFindex);
+  assertEqual(results{3}.HRFindex, expected{3}.HRFindex);
+  assertEqual(results{4}.HRFindex, expected{4}.HRFindex);
```

Comment:
Currently only checking that the HRFindex are equal to what we expect.

- allows to have "lighter" expected data in the `tests/expected` folder
- those are integer values so it is a less "brittle test":  less likely to lead to a test failure compared to a float that might fluctuate with a different MATLAB version in CI or when someone is testings locally.

Let me know if you think think there are other obvious values that should be checked.

Reply:
The HRFindex check is pretty good.  It is actually somewhat-brittle, in the sense that for voxels with no signal (like outside of the brain), the estimated HRFindex might fluctuate a bit.  But in a sense, that is good, because it is stringent.

I reviewed the outputs.  Erring on the side of "checking less / speed", you could consider doing a check on the 4th model output's overall R2 value:

like, results{4}.R2

You could allow some tolerance.  Like, things are ok if    assert(all(abs(expectedR2(:) - results{4}.R2(:)) < 0.1));

Might be good to look at a histogram of the results{4}.R2(:) values that you are getting, just to make sure they are sane.  (And I wonder if there are NaNs, or Infs, or crazy values like that.)

Reply:
```matlab
>> any(isnan(results{4}.R2(:)))

ans =

  logical

   0

>> any(isinf(results{4}.R2(:)))

ans =

  logical

   0
```

`hist(results{4}.R2(:), 100)` gives this (looks reasonable to me)


![untitled](https://user-images.githubusercontent.com/6961185/162251567-43157176-c58e-4f5f-8b8a-c2d49b059de6.jpg)